tests/e2e-playwright/tests/sim4life.py

```diff
@@ -46,39 +61,66 @@ def log_in_and_out(osparc_test_id_attribute: None, context: BrowserContext, page
     yield
 
     print("After test cleaning: Logging out starts")
-    api_request_context = context.request
     api_request_context.post(f"{PRODUCT_URL}v0/auth/logout")
 
 
-def test_billable_sim4life(page: Page, log_in_and_out):
+def test_billable_sim4life(page: Page, log_in_and_out: None, api_request_context: APIRequestContext):
     # connect and listen to websocket
     page.on("websocket", on_web_socket)
 
+    # Welcome to Sim4Life
+    page.wait_for_timeout(5000)
+    welcomeToSim4LifeLocator = page.get_by_text("Welcome to Sim4Life")
+    if welcomeToSim4LifeLocator.is_visible():
+        page.get_by_text("").nth(1).click()  # There is missing osparc-test-id for this button
+    # Quick start window
+    quickStartWindowCloseBtnLocator = page.get_by_test_id("quickStartWindowCloseBtn")
+    if quickStartWindowCloseBtnLocator.is_visible():
+        quickStartWindowCloseBtnLocator.click()
+
     # open services tab and filter for sim4life service
     page.get_by_test_id("servicesTabBtn").click()
     _textbox = page.get_by_role("textbox", name="search")
     _textbox.fill(SERVICE_KEY)
     _textbox.press("Enter")
     page.get_by_test_id(SERVICE_TEST_ID).click()
 
-    # Project detail view pop-ups shows
-    page.get_by_test_id("openResource").click()
-    # Open project with default resources
-    page.get_by_test_id("openWithResources").click()
-    page.wait_for_timeout(5000)
+    with page.expect_response(re.compile(r'/projects/')) as response_info:
+        # Project detail view pop-ups shows
+        page.get_by_test_id("openResource").click()
+        if bool(int(PRODUCT_BILLABLE)):
+            # Open project with default resources
+            page.get_by_test_id("openWithResources").click()
+        page.wait_for_timeout(1000)
+
+    # Get project uuid, will be used to delete this project in the end
+    uuid_pattern = re.compile(r'/projects/([0-9a-fA-F-]+)')
+    match = uuid_pattern.search(response_info.value.url)
+    extracted_uuid = match.group(1)
 
-    # Wait until grid is shown and click on the similation button
+    # Wait until grid is shown
     page.frame_locator(".qx-main-dark").get_by_role("img", name="Remote render").click(
         button="right", timeout=600000
     )
-    page.frame_locator(".qx-main-dark").get_by_text("Simulation").click()
-    page.wait_for_timeout(5000)
+    page.wait_for_timeout(1000)
```

Comment:
Q: what is this used for? Could you have not used `time.sleep` instead? is there a difference?